### PR TITLE
project: Add `merge_pipelines_enabled` and `merge_trains_enabled`. Closes #623

### LINKED ISSUE #623: Support option "Enable merged results pipelines" for projects
I'd like to see support to set the following option via Terraform: "enable merged results pipelines"

Also the feature below, merge trains, would be great to configure from terraform.

image

Settings -> General -> Merge request

### Affected Resource(s)
Please list the resources as a list, for example:
- gitlab_project

### PR DESCRIPTION
**Pull request by jtymes**
## Description
Add `merge_pipelines_enabled` and `merge_trains_enabled`. This allows for managing Merged Results Pipelines and Merge Trains. Closes #623

### PR Checklist Acknowledgement

- [x] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

File: docs/data-sources/project.md [hunk 1]
```diff
@@ -38,7 +38,9 @@ data "gitlab_project" "example" {
 - **http_url_to_repo** (String) URL that can be provided to `git clone` to clone the
 - **issues_enabled** (Boolean) Enable issue tracking for the project.
 - **lfs_enabled** (Boolean) Enable LFS for the project.
+- **merge_pipelines_enabled** (Boolean) Whether merged results pipelines are enabled or not.
```
Comment (timofurrer) [inline docs/data-sources/project.md:41]: Let's use the descriptions from https://docs.gitlab.com/ee/api/projects.html

File: docs/resources/project.md [hunk 2]
```diff
@@ -60,8 +60,10 @@ resource "gitlab_project" "example-two" {
 - **issues_template** (String) Sets the template for new issues in the project.
 - **lfs_enabled** (Boolean) Enable LFS for the project.
 - **merge_method** (String) Set to `ff` to create fast-forward merges
+- **merge_pipelines_enabled** (Boolean) Enable or disable merge pipelines.
 - **merge_requests_enabled** (Boolean) Enable merge requests for the project.
 - **merge_requests_template** (String) Sets the template for new merge requests in the project.
+- **merge_trains_enabled** (Boolean) Enable or disable merge trains. Requires `merge_pipelines_enabled` to be set to `true` to take effect.
```
Comment (timofurrer) [inline docs/resources/project.md:66]: nice comment 🎉

**Review by timofurrer — CHANGES_REQUESTED** [review]
Looks great to me, a nitpick, but in general LGTM 🎉

File: docs/data-sources/project.md (hunk 1 above)
Comment (jtymes) [reply docs/data-sources/project.md:41]: I had that originally, it just read funny to me on a data source since it's not modifying. I'll update.

Comment (jtymes) [reply docs/data-sources/project.md]: Updated 727856f

**Review by timofurrer — DISMISSED** [review]

**Comment by timofurrer** [comment]
@jtymes there are some test failures against a CE instance. The reason for this is that the fields are EE only ...

**Comment by jtymes** [comment]
@timofurrer Whoops, forgot to restart acceptance in CE. Should I just remove those fields from `TestAccGitlabProject_basic`then? Still trying to figure out the structure of all the tests 🙂 .

**Comment by timofurrer** [comment]
@jtymes Yes, in this case that's okay, since you have the additional tests and is basically what is already being done for other flags like `merge_requests_templates`, or `mirror_trigger_builds`, ... :)

**Review by timofurrer — APPROVED** [review]

**Pull request merged**